File: apps/android/feature/ai/domain/src/main/java/com/scribblefit/feature/ai/domain/repository/AnalysisRepository.kt
package com.scribblefit.feature.ai.domain.repository

import com.scribblefit.feature.ai.domain.model.AnalysisSuggestion
import com.scribblefit.feature.ai.domain.model.AnalysisSummary
import com.scribblefit.feature.ai.domain.model.ExerciseInsight
import com.scribblefit.feature.ai.domain.model.SummaryPeriod
import kotlinx.coroutines.flow.Flow

interface AnalysisRepository {
    // Read cached insights
    fun getHomeSuggestion(): Flow<AnalysisSuggestion?>
    fun getSummary(period: SummaryPeriod): Flow<AnalysisSummary?>
    fun getExerciseInsight(exerciseId: String): Flow<ExerciseInsight?>

    // Cache management (called by UseCases after LLM processing)
    suspend fun saveHomeSuggestion(suggestion: AnalysisSuggestion)
    suspend fun saveSummary(summary: AnalysisSummary)
    suspend fun saveExerciseInsight(insight: ExerciseInsight)
    
    // Manual trigger for refresh logic
    suspend fun clearOldInsights()
}

File: apps/ios/ScribbleFit/ScribbleFit/Features/AI/Domain/Repository/AnalysisRepository.swift
import Foundation

public protocol AnalysisRepository: Sendable {
    // Read cached insights
    func getHomeSuggestion() async throws -> AnalysisSuggestion?
    func getSummary(period: SummaryPeriod) async throws -> AnalysisSummary?
    func getExerciseInsight(exerciseId: String) async throws -> ExerciseInsight?

    // Cache management
    func saveHomeSuggestion(_ suggestion: AnalysisSuggestion) async throws
    func saveSummary(_ summary: AnalysisSummary) async throws
    func saveExerciseInsight(_ insight: ExerciseInsight) async throws
    
    func clearOldInsights() async throws
}
